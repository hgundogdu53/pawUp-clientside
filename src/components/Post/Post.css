* {
    box-sizing: border-box;
  }

.flex-container {
    overflow: hidden;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

}

.post-form-section {
    transition: 0.3s;
    border-radius: 5px;
    width: 22.5rem;
    margin-top: 0.5rem;
    background-color: rgba(93, 136, 168, 0.616);
    width: 25rem;
}

.post-form-section:hover {
    box-shadow: 0 8px 16px 0 rgba(255, 255, 255, 0.151);
}


textarea {
  width: 100%;
  height: 150px;
  padding: 12px 20px;
  box-sizing: border-box;
  border: 2px solid #ccc;
  border-radius: 4px;
  background-color: #f8f8f8;
  resize: none;
}

.sub-btn,
.res-btn {
    background-color: #fff;
    color: #004690;
    font-weight: bold;
    text-decoration: none;
    margin-top: 15px;
    padding: 7.5px 20px;
    margin-left: 0.5rem;
    text-transform: uppercase;
    text-decoration: none;
    display: inline-block;
    transition: all .2s;
    border-radius: 100px;
}

.sub-btn:hover,
.res-btn:hover {
    cursor: pointer;
    transform: translateY(-2.5px);
    box-shadow: 0 5px 15px rgba(0, 17, 255, 0.2);
}

.sub-btn:active,
.res-btn:active {
    transform: translateY(-1px);
}

.post-form {
    padding: 0.25rem 1.25rem;
}

.post-form label, .post-form input {
    display: block;
    width: 100%;
    text-align: start;
    margin-top: 0.25rem;
}

.post-form input {
    border: 1px solid #ccc;
    padding: 0.2rem 0.5rem;
    font: inherit;
}

.post-form input:focus {
    outline: none;
    background: rgb(186, 229, 243);
    border-color: rgb(4, 127, 168);
}

.post-form label {
    font-weight: bold;
}

.post-title {
    text-align: center;
    font-size: 1.75rem;
}

.post-form button[type="submit"],
.post-form button[type='reset'] {
    display: inline-block;
    margin-top: 1rem;
}

.post-form-section {
  padding: 10px 35px;
}



@media only screen and (min-width: 40rem) {
    .post-form {
        margin: auto;
        width: 20rem;
    }
}

@media only screen and (min-width: 782px) {
    .post-form-section {
        width: 40rem;
    }
}
