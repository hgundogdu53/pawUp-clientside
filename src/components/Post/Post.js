import React from 'react';
import './Post.css';

function Post(props) {
    return (
        <main className='flex-container'>
            <header>
                <h1 className='post-title'>New Post</h1>
            </header>
            <section className='post-form-section'>
                <form id="post-form" className='post-form' onSubmit={props.addPost}>
                    <div className="form-section">
                        <label htmlFor="pet_name">Pet Name</label>
                        <input type="text" name="pet_name" placeholder="Muffin" required />
                    </div>
                    <div className="form-section">
                        <label htmlFor="type_of_pet">Type of Pet</label>
                        <input type="text" name="type_of_pet" placeholder="Cat" required />
                    </div>
                    <div className="form-section">
                        <label htmlFor="birthdate">Birthdate</label>
                        <input type="date" name="birthdate" />
                    </div>
                    <div className="form-section">
                        <label htmlFor="location">Location</label>
                        <input type="text" name="location" placeholder="San Francisco" />
                    </div>
                    <div className="form-section">
                        <label htmlFor="hobbies">Hobbies</label>
                        <textarea name="hobbies" rows="15" placeholder="Please type in hobbies of your pet" required></textarea>
                    </div>
                    <button type="submit" className="sub-btn">Submit</button>
                    <button type="reset" className="res-btn">Reset</button>
                </form>
            </section>
        </main>
    )
}

export default Post;
