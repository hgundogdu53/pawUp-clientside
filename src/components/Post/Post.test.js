import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Post from './Post';

describe(`Post component`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<Post />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});