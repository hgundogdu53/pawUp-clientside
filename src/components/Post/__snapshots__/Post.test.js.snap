// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component renders without crashing 1`] = `
<main
  className="flex-container"
>
  <header>
    <h1
      className="post-title"
    >
      New Post
    </h1>
  </header>
  <section
    className="post-form-section"
  >
    <form
      className="post-form"
      id="post-form"
    >
      <div
        className="form-section"
      >
        <label
          htmlFor="pet_name"
        >
          Pet Name
        </label>
        <input
          name="pet_name"
          placeholder="Muffin"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="type_of_pet"
        >
          Type of Pet
        </label>
        <input
          name="type_of_pet"
          placeholder="Cat"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="birthdate"
        >
          Birthdate
        </label>
        <input
          name="birthdate"
          type="date"
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="location"
        >
          Location
        </label>
        <input
          name="location"
          placeholder="San Francisco"
          type="text"
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="hobbies"
        >
          Hobbies
        </label>
        <textarea
          name="hobbies"
          placeholder="Please type in hobbies of your pet"
          required={true}
          rows="15"
        />
      </div>
      <button
        className="sub-btn"
        type="submit"
      >
        Submit
      </button>
      <button
        className="res-btn"
        type="reset"
      >
        Reset
      </button>
    </form>
  </section>
</main>
`;
